.github/workflows/core_matrix_build.yml

```diff
@@ -0,0 +1,61 @@
+name: core
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+concurrency:
+  group: ${{ github.head_ref }} || concat(${{ github.ref }}, ${{ github.workflow }})
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        # the result of the matrix will be the combination of all attributes, so we get os*compiler builds
+        include:
+          - os: ubuntu-20.04
+            compiler: clang12
+          - os: ubuntu-20.04
+            compiler: clang11
+          - os: ubuntu-18.04
+            compiler: clang10
+          - os: ubuntu-18.04
+            compiler: gcc8
+          - os: ubuntu-20.04
+            compiler: gcc # default in 20.04 is gcc 9
+          - os: ubuntu-20.04
+            compiler: gcc10
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }}-${{ matrix.compiler }}
+    env:
+      COMPILER: ${{ matrix.compiler }}
+    if: github.repository == 'azerothcore/azerothcore-wotlk' && (github.event.label.name == 'cpp' || github.ref == 'refs/heads/master')
```

Comment:
for some reasons this doesn't work... if I remove the part after `&&` it works, but with the label condition it will never trigger